src/neptune_scale/core/components/aggregating_queue.py

```diff
@@ -68,47 +68,33 @@ def commit(self) -> None:
 
     def get(self) -> BatchedOperations:
         start = time.process_time()
-        elements_in_batch: int = 0
-        batch: Optional[RunOperation] = None
+
+        batch_operations: list[(float, RunOperation)] = []
```

Comment:
Wouldn't it be cleaner to just store the previous `operation_key` in a separate variable? The `float` item is not used anywhere except for comparing the last key, correct?

Reply:
yeah, we don't need to remember previous keys except for the last one
This code seemed slightly safer, since keeping a (key, element) as a structure ensured I wouldn't ever keep an old key while adding an element by mistake in any execution path
but I can remove this safeguard now that it's mostly done